api/envoy/config/route/v3/route_components.proto

```diff
@@ -2192,7 +2224,21 @@ message RateLimit {
   // rate limit configuration. If the override value is invalid or cannot be resolved
   // from metadata, no override is provided. See :ref:`rate limit override
   // <config_http_filters_rate_limit_rate_limit_override>` for more information.
+  //
+  // .. note::
+  //   This is not supported if the rate limit action is configured in the ``typed_per_filter_config`` like
+  //   :ref:`VirtualHost.typed_per_filter_config<envoy_v3_api_field_config.route.v3.VirtualHost.typed_per_filter_config>` or
+  //   :ref:`Route.typed_per_filter_config<envoy_v3_api_field_config.route.v3.Route.typed_per_filter_config>`, etc.
   Override limit = 4;
+
+  // An optional hits addend to be appended to the descriptor produced by this rate limit
```

Comment:
instead of appending to the descriptor,  it should  be used to populate the `hits_addend` field in the RLS request
https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ratelimit/v3/rls.proto#service-ratelimit-v3-ratelimitrequest

Reply:
Single `RateLimit` message is used to generate a descriptor. The the `hits_addend` field here is also a descriptor level configuration. So, the generated value of `hits_addend` should also be populated to the  per-descriptor `hits_addend` rather than request level `hits_addend`.

Reply:
shouldnt this be one level up, under https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-ratelimit ? so it can set the `hits_addend` field of https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/ratelimit/v3/rls.proto#service-ratelimit-v3-ratelimitrequest ? 
The ratelimit request holds a list of descriptors

Reply:
The https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-ratelimit is what I said single `RateLimit`.

Every route/vhost will have a list of `RateLimit` message. Every `Ratelimit` will generate a single descriptor (a single descriptor is composited of multiple descriptor entries). All these descriptor finally composite the rate limit request.

**But note, every descriptor is evaluated independently and will match to completely different rules. One of the most important target of this PR's target is to support per-descriptor hits addend, then we can limit different resource like qps or bandwidth in same request.** You can check this PR's description for the PR's target.